tecken/apps.py

```diff
@@ -96,32 +97,47 @@ def _redis_store_eviction_policy():
 
     @staticmethod
     def _check_storage_urls():
-        """If you use minio to functionally test S3, since it's
+        """Ensure emulated storage buckets exist.
+
+        If you use minio to functionally test S3, since it's
         ephemeral the buckets you create disappear after a restart.
         Make sure they exist. That's what we expect to happen with the
         real production S3 buckets.
+
+        The emulated GCS may also have missing buckets that can be created
+        during initialization.
```

Comment:
We have similar issues in Socorro. I'm of the mind that data storage things should exist before starting Tecken--Tecken should not be required to create them. This reduces the permissions Tecken needs to run.

I think setting this up should be the job of `make setup`.

Does that create new problems and/or doesn't really work in this situation?

Reply:
Also, we could keep this and change it in a future cleanup PR. I don't mean to suggest we have to change it in this one.

Reply:
I think it may be a step too far for this PR. I think either local dev buckets would need to be hard-coded, or bash would need to call Python utilities to parse URLs, and optionally verify and create buckets. Maybe a management command is a compromise? I'll keep it in mind for the storage refactor.

Reply:
We do this in Socorro for the db, es, and pubsub. The way we did it in Socorro is nice because those give developers additional tools for scripting things. It's been really helpful.

For s3, we have an aws cli wrapper. I'd love to rewrite how the s3 one works--the aws cli wrapper is pretty annoying.

Reply:
I think I understand better, having looked at the Socorro implementation, which does a lot more with ``make setup``. I think the way forward would be to move ``_check_storage_urls`` to a management command, and call it from the setup script. That allows us to reuse code in scripts and at runtime, and gets away from "at startup" initialization. Since that changes existing code around minio buckets, I'd like to focus on that in a future cleanup PR.

Reply:
[bug 1556775](https://bugzilla.mozilla.org/show_bug.cgi?id=1556775) is open, so I added [a comment](https://bugzilla.mozilla.org/show_bug.cgi?id=1556775#c7) about the new additions to ``make setup``

Reply:
:+1: